python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -101,12 +106,23 @@ def _execUpdateAction(self):
                 else:
                     action.callback_Failure()
 
+            self._currently_running_command = False
+
 # Create the thread pool
 
 
 def _makeThreadPool(threadPoolSize=THREAD_POOL_SIZE, daemonic=True):
+    global ThreadPool
+    if ThreadPool and len(ThreadPool) != 0:
+        #from Ganga.Core.exceptions import GangaException
+        #raise GangaException("Cannot doubbly init the ThreadPool! ThreadPool already populated with threads")
+        log.debug("Found a thread pool already in existance, wiping it and startig again!")
+        for i in ThreadPool:
+            del i
+        del ThreadPool[:]
+        ThreadPool = []
```

Comment:
Ok, that makes sense. I guess the only thing left is some debug statements notifying if `stop_and_free_thread_pool` did not manage to close all its threads before timing out but that's not something that's needed here.